### PR TITLE
Bots for flamenco tavern

### LINKED ISSUE #442: Liquidity Provisioning Bots that don't burn cash all the time
When [Liquidity Provisioning Orders exist](https://github.com/orgs/vegaprotocol/projects/50), [our current bots](https://github.com/vegaprotocol/traderbot) will no longer be up to the job. We need newer, fancier bots that use LP orders. Probably?

**NOTE** This is not a core spec. The work should not go in the specs folder.

### PR DESCRIPTION
**Pull request by davidsiska-vega**
When finished will close #442

### CODE REVIEW

**Comment by davidsiska-vega** [comment]
@tamlyn10 / @ashleyvega / @edd / @peterbarrow  if any of you would feel like reviewing I would be much obliged.

File: non-protocol-specs/0001-liquidity-providing-bots.md [hunk 1]
```diff
@@ -14,13 +14,191 @@ The downside is that if the bot is "unlucky" they can run out of even large amou
...
+
+### Acceptance criteria
+
+- [ ] Bot can submit a market proposal (optional), commit liquidity and then manage it's position as described above, see also [LP order type](../specs/0038-liquidity-provision-order-type.md).
+- [ ] Bot can connect to an existing market, submit an [LP order type](../specs/0038-liquidity-provision-order-type.md) and then manage it's position as described above.  
+- [ ] Bot can participate in an opening auction placing orders around target price (set via params, see above).
+- [ ] Can read a price target from external source and and places limit orders that "steer" the price up-or-down as appropriate and have the right `targetLNVol` using one of the methods above (note that this has to take into account other identical bots trying to do the same on the same market).
+- [ ] Bot manages its position in such a way that it stays close to zero and starts placing market orders if configured maxima are breached. 
+- [ ] The code can be open sourced. 
+- [ ] Bot is called Bot Mc BotFace.
```
Comment (edd) [inline non-protocol-specs/0001-liquidity-providing-bots.md]:
```suggestion
- [ ] Bot is not called Bot Mc BotFace. 
```

File: non-protocol-specs/0001-liquidity-providing-bots.md [hunk 2]
```diff
@@ -14,13 +14,191 @@ The downside is that if the bot is "unlucky" they can run out of even large amou
...
+Don't use any of the pseudocode above! 
+
+### Acceptance criteria
+
+- [ ] Bot can submit a market proposal (optional), commit liquidity and then manage it's position as described above, see also [LP order type](../specs/0038-liquidity-provision-order-type.md).
+- [ ] Bot can connect to an existing market, submit an [LP order type](../specs/0038-liquidity-provision-order-type.md) and then manage it's position as described above.  
+- [ ] Bot can participate in an opening auction placing orders around target price (set via params, see above).
+- [ ] Can read a price target from external source and and places limit orders that "steer" the price up-or-down as appropriate and have the right `targetLNVol` using one of the methods above (note that this has to take into account other identical bots trying to do the same on the same market).
+- [ ] Bot manages its position in such a way that it stays close to zero and starts placing market orders if configured maxima are breached. 
+- [ ] The code can be open sourced.
```
Comment (edd) [inline non-protocol-specs/0001-liquidity-providing-bots.md]:
```suggestion
- [ ] The repository is public from the start. 
```

**Comment by edd** [comment]
Seems good to me. I think we should merge this and plan building it. 

One area that is not fully covered is *how* a market decides to open a market. The process of opening a market is covered, but not what would trigger a bot to do so. That seems fine - let's just know that should be possible and build that as an addition.

**Review by edd — APPROVED** [review]

**Comment by tamlyn10** [comment]
Are you talking about where it says "create markets"? I think that David probably means that in the sense of making liquidity not proposing a market.

**Comment by edd** [comment]
> Are you talking about where it says "create markets"? I think that David probably means that in the sense of making liquidity not proposing a market.

No. See "Submitting a market proposal"

**Comment by davidsiska-vega** [comment]
> > Are you talking about where it says "create markets"? I think that David probably means that in the sense of making liquidity not proposing a market.
> 
> No. See "Submitting a market proposal"

So to submit a market proposal I just mean that the input for the bot is:
- all that is needed to propose a market (so underlying, expiry, risk model, price monitoring, oracle ... the works)
- the liquidity shapes and commitment amount (fraction of total cash)

The bot will then submit the market proposal, with liquidity commitment and run with it.

The intention is *not* that the bot will be smart enough to decide which market to propose and how the market should look.

**Review by witgaw — COMMENTED** [review]
@davidsiska-vega the spec looks good to me, just slightly confused by:
 "serve as a liquidity provider for the market by submitting the [LP order type](../specs/0038-liquidity-provision-order-type.md) **(optional)**." and
`Serving as a liquidity provider for a market` and `Create markets that look real` sections.

Is it that both sections talk about the same type of bot, just emphasise different aspects of its behaviour or will we have two types of bots:
- "registered" liquidity providers (submit LP commitment, use LP orders, get the share of the rewards) [`Serving as a liquidity provider for a market` section] and
- "exetnal/effecitve" liquidity providers which try to make money by acting as a market maker, but use only limit orders and don't get a share of the rewards except for the maker fee  [`Create markets that look real` section]?

**Comment by davidsiska-vega** [comment]
> @davidsiska-vega the spec looks good to me, just slightly confused by:
> "serve as a liquidity provider for the market by submitting the [LP order type](../specs/0038-liquidity-provision-order-type.md) **(optional)**." and
> `Serving as a liquidity provider for a market` and `Create markets that look real` sections.
> 
> Is it that both sections talk about the same type of bot, just emphasise different aspects of its behaviour or will we have two types of bots:
> 
> * "registered" liquidity providers (submit LP commitment, use LP orders, get the share of the rewards) [`Serving as a liquidity provider for a market` section] and
> * "exetnal/effecitve" liquidity providers which try to make money by acting as a market maker, but use only limit orders and don't get a share of the rewards except for the maker fee  [`Create markets that look real` section]?

The idea is to have one codebase for bot that configurable to do either (or both) of the above. The sections then explain what it entails if the option is enabled. If that part is switched off the section should be ignored.

**Comment by davidsiska-vega** [comment]
@witgaw I think I pushed the clarifications you were seeking with your comments. Please look.

**Review by witgaw — APPROVED** [review]
Thanks for adding the clarifications, looks great to me now.

**Comment by davidsiska-vega** [comment]
@tamlyn10 did you want to review it before I merge? Ditto at @ashleyvega - I think it's ready...

**Comment by tamlyn10** [comment]
I'm ok with it..

**Comment by davidsiska-vega** [comment]
I'll merge as @ashleyvega will have to look at the spec anyway while implementing it so he can catch any nonsense that way and potential changes to the spec shouldn't then surprise him, if he's the one requested them ;-).

**Pull request merged**